pkg/devfile/generator/utils.go

```diff
@@ -153,11 +153,20 @@ func addSyncRootFolder(container *corev1.Container, sourceMapping string) string
 
 	// Note: PROJECTS_ROOT & PROJECT_SOURCE are validated at the devfile parser level
 	// Add PROJECTS_ROOT to the container
-	container.Env = append(container.Env,
-		corev1.EnvVar{
+	container.Env = append([]corev1.EnvVar{
+		{
 			Name:  EnvProjectsRoot,
 			Value: syncRootFolder,
-		})
+		},
+	}, container.Env...)
+
+	if sourceMapping != "" {
```

Comment:
We don't need this check in `addSyncRootFolder` we only need to add `EnvProjectsRoot` which you are doing above.